### PR TITLE
Update service refactor

### PR DESCRIPTION
**Pull request by SoManyHs**
See `cd98949`

Includes refactors to service test to remove unnecessary helpers/mocks.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-cli/modules/cli/compose/entity/service/service_test.go [hunk 1]
```diff
@@ -864,108 +894,37 @@ func upServiceWithCurrentTaskDefTest(t *testing.T,
...
-	}
-	return emptyDescribeServiceResponse
 }
 
 func getTestTaskDef(taskDefID string) (taskDefArn string, taskDefinition, registerTaskDefResponse ecs.TaskDefinition) {
-	taskDefArn = "arn/" + taskDefID
+	taskDefArn = arnPrefix + taskDefID
 	taskDefinition = ecs.TaskDefinition{
-		Family:               aws.String("family"),
+		Family:               aws.String(taskDefID),
```
[the remark this answers is not in the document]
Comment (SoManyHs) [reply ecs-cli/modules/cli/compose/entity/service/service_test.go]: In all of our tests it doesn't appear that we include the revision. The problem with just having `"family"` here previously was that this didn't match the actual family that the tests were using (see `updateServiceTest` or `createServiceTest`)

[the remark this answers is not in the document]
Comment (SoManyHs) [reply ecs-cli/modules/cli/compose/entity/service/service_test.go]: hm, in the spirit of making `getTestTaskDef` more realistic, should I add a revision to the `taskDefArn` that we construct then?

[the remark this answers is not in the document]
Comment (allisaurus) [reply ecs-cli/modules/cli/compose/entity/service/service_test.go]: sure, that would indeed be a better match for the actual ARN

**Comment by SoManyHs** [comment]
# Current behavior
```
$ ecs-cli compose --ecs-params ecs_params/ecs-params-sd.yml service up --cluster test
INFO[0000] Using ECS task definition                     TaskDefinition="test_fields:64"
INFO[0000] Created an ECS service                        service=test_fields taskDefinition="test_fields:64"
INFO[0000] Updated ECS service successfully              desiredCount=1 serviceName=test_fields
INFO[0015] (service test_fields) has started 1 tasks: (task a6a21063-5907-4f76-81ca-686f61dd9426).  timestamp="2018-12-04 22:19:05 +0000 UTC"
INFO[0030] Service status                                desiredCount=1 runningCount=1 serviceName=test_fields
INFO[0030] ECS Service has reached a stable state        desiredCount=1 runningCount=1 serviceName=test_fields

$ ecs-cli compose --ecs-params ecs_params/ecs-params-sd.yml service down --cluster test
INFO[0005] Updated ECS service successfully              desiredCount=0 serviceName=test_fields
INFO[0005] Service status                                desiredCount=0 runningCount=1 serviceName=test_fields
INFO[0020] Service status                                desiredCount=0 runningCount=0 serviceName=test_fields
INFO[0020] (service test_fields) has stopped 1 running tasks: (task a6a21063-5907-4f76-81ca-686f61dd9426).  timestamp="2018-12-04 22:29:24 +0000 UTC"
INFO[0020] ECS Service has reached a stable state        desiredCount=0 runningCount=0 serviceName=test_fields
INFO[0020] Deleted ECS service                           service=test_fields
INFO[0020] ECS Service has reached a stable state        desiredCount=0 runningCount=0 serviceName=test_fields

```

# New behavior
```
~/ecs/ecs-cli/testing/projects/test_fields(master)$ ecs-cli compose --ecs-params ecs_params/ecs-params-sd.yml service up --cluster bork
INFO[0000] Using ECS task definition                     TaskDefinition="test_fields:64"
INFO[0000] Created an ECS service                        service=test_fields taskDefinition="test_fields:64"
INFO[0000] Updated ECS service successfully              desiredCount=1 force-deployment=false service=test_fields
INFO[0015] (service test_fields) has started 1 tasks: (task e9e8cb48-5d2c-408c-87ed-3b2dc1280a97).  timestamp="2018-12-04 22:20:21 +0000 UTC"
INFO[0030] Service status                                desiredCount=1 runningCount=1 serviceName=test_fields
INFO[0030] (service test_fields) has reached a steady state.  timestamp="2018-12-04 22:20:40 +0000 UTC"
INFO[0030] ECS Service has reached a stable state        desiredCount=1 runningCount=1 serviceName=test_fields

$ ecs-cli compose --ecs-params ecs_params/ecs-params-sd.yml service down --cluster bork
INFO[0000] Updated ECS service successfully              desiredCount=0 force-deployment=false service=test_fields
INFO[0000] Service status                                desiredCount=0 runningCount=1 serviceName=test_fields
INFO[0015] Service status                                desiredCount=0 runningCount=0 serviceName=test_fields
INFO[0015] (service test_fields) has stopped 1 running tasks: (task e9e8cb48-5d2c-408c-87ed-3b2dc1280a97).  timestamp="2018-12-04 22:21:39 +0000 UTC"
INFO[0015] ECS Service has reached a stable state        desiredCount=0 runningCount=0 serviceName=test_fields
INFO[0015] Deleted ECS service                           service=test_fields
INFO[0015] ECS Service has reached a stable state        desiredCount=0 runningCount=0 serviceName=test_fields
```

**Pull request merged**